core/domain/feedback_services.py

```diff
@@ -368,7 +421,123 @@ def _get_thread_from_model(thread_model):
         thread_model.id, thread_model.exploration_id, thread_model.state_name,
         thread_model.original_author_id, thread_model.status,
         thread_model.subject, thread_model.summary, thread_model.has_suggestion,
-        thread_model.created_on, thread_model.last_updated)
+        thread_model.message_count, thread_model.created_on,
+        thread_model.last_updated)
+
+
+def set_message_count_for_thread(full_thread_id, message_count):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count = message_count
+    thread_model.put()
+
+
+def increase_message_count_by_one(full_thread_id):
+    """Assigns the message count field the number of messages present in the
+    thread.
+
+    Args:
+        full_thread_id: str. The full id of the thread.
+        message_count: int. The number of messages in the thread.
+    """
+    thread_model = feedback_models.FeedbackThreadModel.get(full_thread_id)
+    thread_model.message_count += 1
+    thread_model.put()
+
+
+def get_thread_summaries(user_id, full_thread_ids):
+    """Returns a list of summaries corresponding to each of the threads given.
+
+    Args:
```

Comment:
Done! Sorry for this, I had earlier followed a completely different approach everywhere but realized later it wouldn't work. So I changed the implementation but forgot to update the docstrings. I will take care next time.